andela/travel_tool_back
# #161340859 Fix user can edit other user's profile

#### What does this PR do?
This PR ensures that a user cannot update other user's profile details

#### Description of Task to be completed?
- Add new test cases for the bugfix
- Create a middleware to validate signed in user

#### How should this be manually tested?
Steps to reproduce

- Log in to https://travela-staging.andela.com/ and obtain your token from the Cookie stored in your browser.
- Add another user other than yourself on the Users table
- Start the backend of the application
- Open Postman and hit the endpoint `PUT /user/:id/profile` with the request body 'gender, occupation, manager' using the other user's id in the id param. Ensure the Authorization header is set with your token.
- The other user's profile is updated

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#161340859](https://www.pivotaltracker.com/story/show/161340859)

#### Screenshots (if appropriate)
<img width="1027" alt="screen shot 2018-10-25 at 7 26 38 am" src="https://user-images.githubusercontent.com/32720508/47486068-6b4fe880-d837-11e8-9090-5eecfcf4b924.png">

#### Questions:
N/A
